### ISSUE TITLE
Sequelize Typescript Issue with Timestamps

### ISSUE THREAD

**Issue by WarienX** [RobinBuschmann/sequelize-typescript#977]
Hello everyone, I am working with sequelize typescript for Postgresql DB. I am having issue with the @CreatedAt and @UpdatedAt tags wherein the table gets created with the createdat and updatedat fields but when Inserting into the table the 2 columns do not get the timestamps assigned to them automatically.

Any Help would be great, thanks in advance.

![roles1](https://user-images.githubusercontent.com/26925053/114357555-7d50ad80-9b8f-11eb-813e-2d5dd0ee096f.PNG)
![roles2DB](https://user-images.githubusercontent.com/26925053/114357560-7e81da80-9b8f-11eb-9ea0-d56f4ef67cdc.PNG)

**Comment by lukashroch** [comment]
Sequelize handles timestamps in code logic. So if you're trying to insert data directly into the DB, those columns need to have correct default values or provide the values in query.

Your model/migration for that table probably doesn't have default value for those columns. Depends how you handle migrations. You can define it on Model-level, which is OK for development/testing - that will generate tables based on defined model.
In PG case, you can try setting default value something line this. These default values often varies based on dialects, hence the literal, it will be different e.g. for mysql, which has CURRENT_TIMESTAMP or something.
```
@Column({
    defaultValue: sequelize.literal("now()"),
    // or defaultValue: sequelize.literal("(now() at time zone 'utc')"),
  })
```

For production, define the default value in migration file, similarly as on model, above.

**Comment by WarienX** [comment]
Thanks @lukashroch, but this syntax "defaultValue: sequelize.literal("now()")" is it applicable for both CreatedAt & UpdatedAt?

**Comment by lukashroch** [comment]
Yeah, it just sets the default value for column, when it's omitted in raw query. You might just double check the syntax in general, it's juts from top of my head, but I think pg has `now()`.

Though all the logic for timestamps updates is in sequelize, this really just adds defaults to schema columns.

**Comment by WarienX** [comment]
Thanks @lukashroch, i'll try it out.
